IdentitySdkCore/IdentitySdkCore/Classes/ReachFiveFidoClient.swift

```diff
@@ -0,0 +1,179 @@
+       
+             self.userConsentUI = UserConsentUI(viewController: self.viewController)
+             self.userConsentUI.config.showRPInformation = false
+             self.userConsentUI.config.fieldTextLightColor = UIColor.black
+             let authenticator = InternalAuthenticator(ui: self.userConsentUI)
+             self.webAuthnClient = WebAuthnClient(
+                 origin:        self.origin,
+                 authenticator: authenticator
+             )
+         }
+    func startRegistration(registrationOption: RegistrationOptions, completion: @escaping ((WebauthnSignupCredential) -> Any)) {
+         
+           var challenge = registrationOption.options.publicKey.challenge
+           let userId = registrationOption.options.publicKey.user.id
+           let userName = registrationOption.options.publicKey.user.name
+           let displayName = registrationOption.options.publicKey.user.displayName
+           let rpId = registrationOption.options.publicKey.rp.id
+          
+           let requireResidentKey = true
+           var options = PublicKeyCredentialCreationOptions()
+           challenge = base64urlToHexString(base64url: challenge)
+          
+           options.challenge = Bytes.fromHex(challenge)
+           options.user.id = Bytes.fromString(userId)
+           options.user.name = userName
+           options.user.displayName = displayName
+           options.rp.id = rpId
+           options.rp.name = rpId
+          
+           options.attestation = .direct
+           options.addPubKeyCredParam(alg: .es256)
+           options.authenticatorSelection = AuthenticatorSelectionCriteria(
+               requireResidentKey: requireResidentKey,
+               userVerification: UserVerificationRequirement.required
+           )
+
+           firstly {
+               self.webAuthnClient.create(options)
+           }.done { credential in
+               
+               let rawId = credential.rawId.toHexString()
+               let credId = credential.id
+               let credType = credential.type.rawValue
+               let clientDataJSON = self.encodeClientDataJsonToBase64(clientDataJson: credential.response.clientDataJSON)
+               let attestationObject = Base64.encodeBase64URL(credential.response.attestationObject)
+
+               let r5AuthenticatorAttestationResponse = R5AuthenticatorAttestationResponse(attestationObject: attestationObject,clientDataJSON: clientDataJSON)
+               
+               let registrationPublicKeyCredential = RegistrationPublicKeyCredential(id: credId,rawId: rawId,type: credType,response: r5AuthenticatorAttestationResponse)
+               
+               let webauthnSignupCredential = WebauthnSignupCredential (webauthnId: userId,publicKeyCredential: registrationPublicKeyCredential)
+               _ = completion(webauthnSignupCredential)
+
+           }.catch { error in
+               print("error : ",error.localizedDescription)
```

Comment:
Will this swallow the error that should instead be thrown for the integrator to catch or handle?

Reply:
For synchronous calls like this method a `Result<T, E>` could be returned instead so the error handling should be done by the caller. For async calls `Future<T, E>` should be used.